apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/ValidatorApp.scala

```
@@ -683,6 +700,45 @@ class ValidatorApp(
     )
   }
 
+  private def persistScanUrlListBuilder(
+      validatorConfigProvider: ValidatorConfigProvider
+  )(implicit traceContext: TraceContext): Seq[DsoScan] => Future[Unit] = {
+    (connections: Seq[DsoScan]) =>
+      {
+        val internalConfigs: Seq[ScanUrlInternalConfig] = connections.map { dsoScan =>
+          ScanUrlInternalConfig(
+            svName = dsoScan.svName,
+            url = dsoScan.publicUrl.toString,
+          )
+        }
+        validatorConfigProvider.setScanUrlInternalConfig(internalConfigs)
+      }
+  }
+
+  private def getPersistedScanList(
+      validatorConfigProvider: ValidatorConfigProvider
+  )(implicit traceContext: TraceContext): Option[Seq[DsoScan]] = {
+    val futureOption: Future[Option[Seq[ScanUrlInternalConfig]]] =
+      validatorConfigProvider.getScanUrlInternalConfig().value
+
+    val maybeInternalConfigs: Option[Seq[ScanUrlInternalConfig]] =
+      try {
+        Await.result(futureOption, 10.seconds) // what is the max waiting time?
+      } catch {
+        case e: Throwable =>
```

Let's avoid catch all scenarios. 
Note: in scala catching throwable would also catch scala flow control exceptions such as return. Worst case scenario we should use catch NonFatal. Also we should never silently ignore exceptions

---

Thanks

Since I removed the Await, i no longer have this catch clause